src/main/java/net/starschema/clouddb/jdbc/BQConnection.java

```diff
@@ -114,16 +115,14 @@ public BQConnection(String url, Properties loginProp) throws SQLException {
             Matcher matchData = projectAndDatasetMatcher.matcher(pathParams);
 
             if (matchData.find()) {
-                this.projectId = matchData.group(1);
+                this.projectId = CatalogName.toProjectId(matchData.group(1));
```

Comment:
👍

Reply:
Neat, so we are _explicitly_ un-mangling here now 👍